container/memory_store.go

```diff
@@ -25,8 +28,13 @@ func (c *memoryStore) Add(id string, cont *Container) {
 
 // Get returns a container from the store by id.
 func (c *memoryStore) Get(id string) *Container {
+	var res *Container
 	c.RLock()
-	res := c.s[id]
+	if runtime.GOOS == "solaris" {
+		res = c.s[id[:12]]
```

Comment:
This feels a little weird… on solaris, container are stored by *only* the first 12 digits ?